### ISSUE TITLE
CH:AllergyIntolerance Example Validation Problem

### ISSUE THREAD

**Issue by oliveregger** [ahdis/matchbox#251]
https://fhir.ch/ig/ch-allergyintolerance/Bundle-ch-allergyintolerance-document-example.json.html

Validate Bundle against http://fhir.ch/ig/ch-allergyintolerance/StructureDefinition/ch-allergyintolerance-document-epr|3.0.0-ballotValidate Composition against http://fhir.ch/ig/ch-allergyintolerance/StructureDefinition/ch-allergyintolerance-composition-epr|3.0.0-ballot..........20..........40..........60..........80..........100..........120..........140..........160..........180..........200..........220..........240..........260..........280......Error from http://tx.fhir.org/r4: Element name mismatch (start: "td"/ end: "t") at Line 1, Column 246982 [143-2443479]

```json
{
    "resourceType": "OperationOutcome",
    "id": "OperationOutcome/0ea27b6986062166eb2eaecaafea7f7b32e1e07bb873296a07969e3f506c4947",
    "issue": [
        {
            "extension": [
                {
                    "extension": [
                        {
                            "url": "profile",
                            "valueUri": "http://fhir.ch/ig/ch-allergyintolerance/StructureDefinition/ch-allergyintolerance-document-epr"
                        },
                        {
                            "url": "profileVersion",
                            "valueUri": "3.0.0-ballot"
                        },
                        {
                            "url": "profileDate",
                            "valueDateTime": "2024-05-17T08:05:32+00:00"
                        },
                        {
                            "url": "total",
                            "valueDuration": {
                                "value": 19366,
                                "unit": "ms"
                            }
                        },
                        {
                            "url": "validatorVersion",
                            "valueString": "powered by matchbox 3.8.5, hapi-fhir 7.0.2 and org.hl7.fhir.core 6.3.5"
                        },
                        {
                            "url": "ig",
                            "valueString": "ch.fhir.ig.ch-allergyintolerance#3.0.0-ballot"
                        },
                        {
                            "url": "hintAboutNonMustSupport",
                            "valueBoolean": false
                        },
                        {
                            "url": "recursive",
                            "valueBoolean": false
                        },
                        {
                            "url": "showMessagesFromReferences",
                            "valueBoolean": false
                        },
                        {
                            "url": "doDebug",
                            "valueBoolean": false
                        },
                        {
                            "url": "assumeValidRestReferences",
                            "valueBoolean": false
                        },
                        {
                            "url": "canDoNative",
                            "valueBoolean": false
                        },
                        {
                            "url": "noExtensibleBindingMessages",
                            "valueBoolean": false
                        },
                        {
                            "url": "noUnicodeBiDiControlChars",
                            "valueBoolean": false
                        },
                        {
                            "url": "noInvariants",
                            "valueBoolean": false
                        },
                        {
                            "url": "displayIssuesAreWarnings",
                            "valueBoolean": true
                        },
                        {
                            "url": "wantInvariantsInMessages",
                            "valueBoolean": false
                        },
                        {
                            "url": "doImplicitFHIRPathStringConversion",
                            "valueBoolean": false
                        },
                        {
                            "url": "securityChecks",
                            "valueBoolean": false
                        },
                        {
                            "url": "crumbTrails",
                            "valueBoolean": false
                        },
                        {
                            "url": "forPublication",
                            "valueBoolean": false
                        },
                        {
                            "url": "httpReadOnly",
                            "valueBoolean": false
                        },
                        {
                            "url": "allowExampleUrls",
                            "valueBoolean": false
                        },
                        {
                            "url": "txServer",
                            "valueUri": "http://tx.fhir.org"
                        },
                        {
                            "url": "fhirVersion",
                            "valueString": "4.0.1"
                        },
                        {
                            "url": "ig",
                            "valueString": "ch.fhir.ig.ch-allergyintolerance#3.0.0-ballot"
                        },
                        {
                            "url": "locale",
                            "valueString": "English"
                        },
                        {
                            "url": "jurisdiction",
                            "valueString": "urn:iso:std:iso:3166#CH"
                        },
                        {
                            "url": "extensions",
                            "valueString": "any"
                        },
                        {
                            "url": "sessionId",
                            "valueString": "-1738073289"
                        },
                        {
                            "url": "package",
                            "valueString": "hl7.fhir.xver-extensions#4.0"
                        },
                        {
                            "url": "package",
                            "valueString": "hl7.fhir.r4.core#4.0.1"
                        },
                        {
                            "url": "package",
                            "valueString": "hl7.fhir.uv.extensions.r4#1.0.0"
                        },
                        {
                            "url": "package",
                            "valueString": "hl7.terminology#5.4.0"
                        },
                        {
                            "url": "package",
                            "valueString": "ihe.formatcode.fhir#1.2.0"
                        },
                        {
                            "url": "package",
                            "valueString": "ch.fhir.ig.ch-term#3.0.0"
                        },
                        {
                            "url": "package",
                            "valueString": "ch.fhir.ig.ch-core#5.0.0-ballot"
                        },
                        {
                            "url": "package",
                            "valueString": "ch.fhir.ig.ch-term#3.0.0"
                        },
                        {
                            "url": "package",
                            "valueString": "fhir.dicom#2022.4.20221006"
                        },
                        {
                            "url": "package",
                            "valueString": "hl7.fhir.uv.ips#1.1.0"
                        },
                        {
                            "url": "package",
                            "valueString": "ch.fhir.ig.ch-allergyintolerance#3.0.0-ballot"
                        }
                    ],
                    "url": "http://matchbox.health/validation"
                }
            ],
            "severity": "information",
            "code": "informational",
            "diagnostics": "Validation for profile http://fhir.ch/ig/ch-allergyintolerance/StructureDefinition/ch-allergyintolerance-document-epr|3.0.0-ballot (2024-05-17T08:05:32+00:00). Loaded packages: hl7.fhir.xver-extensions#4.0, hl7.fhir.r4.core#4.0.1, hl7.fhir.uv.extensions.r4#1.0.0, hl7.terminology#5.4.0, ihe.formatcode.fhir#1.2.0, ch.fhir.ig.ch-term#3.0.0, ch.fhir.ig.ch-core#5.0.0-ballot, ch.fhir.ig.ch-term#3.0.0, fhir.dicom#2022.4.20221006, hl7.fhir.uv.ips#1.1.0, ch.fhir.ig.ch-allergyintolerance#3.0.0-ballot. Duration: 19.366s. powered by matchbox 3.8.5, hapi-fhir 7.0.2 and org.hl7.fhir.core 6.3.5. Validation parameters: CliContext{doNative=false, extensions=[any], hintAboutNonMustSupport=false, recursive=false, showMessagesFromReferences=false, doDebug=false, assumeValidRestReferences=false, canDoNative=false, noExtensibleBindingMessages=false, noUnicodeBiDiControlChars=false, noInvariants=false, displayIssuesAreWarnings=true, wantInvariantsInMessages=false, doImplicitFHIRPathStringConversion=false, htmlInMarkdownCheck=WARNING, txServer='http://tx.fhir.org', lang='null', snomedCT='null', fhirVersion='4.0.1', ig='ch.fhir.ig.ch-allergyintolerance#3.0.0-ballot', questionnaireMode=CHECK, level=HINTS, mode=VALIDATION, securityChecks=false, crumbTrails=false, forPublication=false, allowExampleUrls=false, locale='English', locations={}, jurisdiction='urn:iso:std:iso:3166#CH', igsPreloaded=null, onlyOneEngine=false, httpReadOnly=false}"
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 4107
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "InstanceValidator"
                }
            ],
            "severity": "information",
            "code": "informational",
            "diagnostics": "This element does not match any known slice defined in the profile http://fhir.ch/ig/ch-allergyintolerance/StructureDefinition/ch-allergyintolerance-document-epr|3.0.0-ballot (this may not be a problem, but you should check that it's not intended to match a slice) Slice info: 1.) Bundle.entry[0]: Does not match slice 'Composition' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-allergyintolerance/StructureDefinition/ch-allergyintolerance-composition-epr')) 2.) Bundle.entry[0]: Bundle.entry[0].resource.section[0].entry[0]: Unable to find a match for profile AllergyIntolerance/AllergyToCatDander among choices: http://fhir.ch/ig/ch-allergyintolerance/StructureDefinition/ch-allergyintolerance 3.) Bundle.entry[0]: Bundle.entry[0].resource.section[0].entry[0]: Bundle.entry[7].resource/*AllergyIntolerance/AllergyToCatDander*/.code: Error from http://tx.fhir.org/r4: Element name mismatch (start: "td"/ end: "t") at Line 1, Column 246982 4.) Bundle.entry[0]: Does not match slice 'Patient' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-patient')) 5.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Patient' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-patient|5.0.0-ballot', but found type 'Composition' 6.) Bundle.entry[0]: Does not match slice 'Practitioner' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-practitioner')) 7.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Practitioner' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-practitioner|5.0.0-ballot', but found type 'Composition' 8.) Bundle.entry[0]: Does not match slice 'PractitionerRole' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-practitionerrole')) 9.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'PractitionerRole' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-practitionerrole|5.0.0-ballot', but found type 'Composition' 10.) Bundle.entry[0]: Does not match slice 'Organization' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-organization')) 11.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Organization' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-organization|5.0.0-ballot', but found type 'Composition' 12.) Bundle.entry[0]: Does not match slice 'RelatedPerson' (discriminator: resource.conformsTo('http://hl7.org/fhir/StructureDefinition/RelatedPerson')) 13.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'RelatedPerson' in profile 'http://hl7.org/fhir/StructureDefinition/RelatedPerson|4.0.1', but found type 'Composition' 14.) Bundle.entry[0]: Does not match slice 'Device' (discriminator: resource.conformsTo('http://hl7.org/fhir/StructureDefinition/Device')) 15.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Device' in profile 'http://hl7.org/fhir/StructureDefinition/Device|4.0.1', but found type 'Composition' 16.) Bundle.entry[0]: Does not match slice 'Encounter' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-encounter')) 17.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Encounter' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-encounter|5.0.0-ballot', but found type 'Composition' 18.) Bundle.entry[0]: Does not match slice 'Location' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-location')) 19.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Location' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-location|5.0.0-ballot', but found type 'Composition' 20.) Bundle.entry[0]: Does not match slice 'Binary' (discriminator: resource.conformsTo('http://hl7.org/fhir/StructureDefinition/Binary')) 21.) Bundle.entry[0]: Bundle.entry[0].resource: Specified profile type was 'Binary' in profile 'http://hl7.org/fhir/StructureDefinition/Binary|4.0.1', but found type 'Composition'",
            "expression": [
                "Bundle.entry[0]"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 14893
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "InstanceValidator"
                }
            ],
            "severity": "information",
            "code": "informational",
            "diagnostics": "This element does not match any known slice defined in the profile http://fhir.ch/ig/ch-allergyintolerance/StructureDefinition/ch-allergyintolerance-document-epr|3.0.0-ballot (this may not be a problem, but you should check that it's not intended to match a slice) Slice info: 1.) Bundle.entry[7]: Does not match slice 'Composition' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-allergyintolerance/StructureDefinition/ch-allergyintolerance-composition-epr')) 2.) Bundle.entry[7]: Bundle.entry[7].resource: Specified profile type was 'Composition' in profile 'http://fhir.ch/ig/ch-allergyintolerance/StructureDefinition/ch-allergyintolerance-composition-epr|3.0.0-ballot', but found type 'AllergyIntolerance' 3.) Bundle.entry[7]: Does not match slice 'Patient' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-patient')) 4.) Bundle.entry[7]: Bundle.entry[7].resource: Specified profile type was 'Patient' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-patient|5.0.0-ballot', but found type 'AllergyIntolerance' 5.) Bundle.entry[7]: Does not match slice 'Practitioner' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-practitioner')) 6.) Bundle.entry[7]: Bundle.entry[7].resource: Specified profile type was 'Practitioner' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-practitioner|5.0.0-ballot', but found type 'AllergyIntolerance' 7.) Bundle.entry[7]: Does not match slice 'PractitionerRole' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-practitionerrole')) 8.) Bundle.entry[7]: Bundle.entry[7].resource: Specified profile type was 'PractitionerRole' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-practitionerrole|5.0.0-ballot', but found type 'AllergyIntolerance' 9.) Bundle.entry[7]: Does not match slice 'Organization' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-organization')) 10.) Bundle.entry[7]: Bundle.entry[7].resource: Specified profile type was 'Organization' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-organization|5.0.0-ballot', but found type 'AllergyIntolerance' 11.) Bundle.entry[7]: Does not match slice 'RelatedPerson' (discriminator: resource.conformsTo('http://hl7.org/fhir/StructureDefinition/RelatedPerson')) 12.) Bundle.entry[7]: Bundle.entry[7].resource: Specified profile type was 'RelatedPerson' in profile 'http://hl7.org/fhir/StructureDefinition/RelatedPerson|4.0.1', but found type 'AllergyIntolerance' 13.) Bundle.entry[7]: Does not match slice 'Device' (discriminator: resource.conformsTo('http://hl7.org/fhir/StructureDefinition/Device')) 14.) Bundle.entry[7]: Bundle.entry[7].resource: Specified profile type was 'Device' in profile 'http://hl7.org/fhir/StructureDefinition/Device|4.0.1', but found type 'AllergyIntolerance' 15.) Bundle.entry[7]: Does not match slice 'Encounter' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-encounter')) 16.) Bundle.entry[7]: Bundle.entry[7].resource: Specified profile type was 'Encounter' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-encounter|5.0.0-ballot', but found type 'AllergyIntolerance' 17.) Bundle.entry[7]: Does not match slice 'Location' (discriminator: resource.conformsTo('http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-location')) 18.) Bundle.entry[7]: Bundle.entry[7].resource: Specified profile type was 'Location' in profile 'http://fhir.ch/ig/ch-core/StructureDefinition/ch-core-location|5.0.0-ballot', but found type 'AllergyIntolerance' 19.) Bundle.entry[7]: Does not match slice 'Binary' (discriminator: resource.conformsTo('http://hl7.org/fhir/StructureDefinition/Binary')) 20.) Bundle.entry[7]: Bundle.entry[7].resource: Specified profile type was 'Binary' in profile 'http://hl7.org/fhir/StructureDefinition/Binary|4.0.1', but found type 'AllergyIntolerance'",
            "expression": [
                "Bundle.entry[7]"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 14895
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "InstanceValidator"
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-message-id",
                    "valueCode": "Validation_VAL_Profile_Minimum_SLICE"
                }
            ],
            "severity": "error",
            "code": "structure",
            "diagnostics": "Slice 'Bundle.entry:Composition': minimum required = 1, but only found 0 (from http://fhir.ch/ig/ch-allergyintolerance/StructureDefinition/ch-allergyintolerance-document-epr|3.0.0-ballot)",
            "expression": [
                "Bundle"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 4039
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "TerminologyEngine"
                }
            ],
            "severity": "warning",
            "code": "invalid",
            "diagnostics": "Der Displayname f\\u00fcr http://loinc.org#48765-2 sollte einer von ''Allergien und unerwünschte Wirkungen - Dokument' (de-DE)' anstelle von 'Allergies and adverse reactions Document' sein.",
            "expression": [
                "Bundle.entry[0].resource/*Composition/AllergyIntoleranceComposition*/.section[0].code.coding[0].display"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 14211
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "TerminologyEngine"
                }
            ],
            "severity": "information",
            "code": "invalid",
            "diagnostics": "The code '232346004' is valid but is not active",
            "expression": [
                "Bundle.entry[7].resource/*AllergyIntolerance/AllergyToCatDander*/.code.coding[0].code"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 14519
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "TerminologyEngine"
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-message-id",
                    "valueCode": "MSG_DRAFT"
                }
            ],
            "severity": "information",
            "code": "business-rule",
            "diagnostics": "Reference to draft CodeSystem http://terminology.hl7.org/CodeSystem/reaction-event-certainty|4.0.1",
            "expression": [
                "Bundle.entry[7].resource/*AllergyIntolerance/AllergyToCatDander*/.reaction[0].extension[0].value.ofType(CodeableConcept)"
            ]
        },
        {
            "extension": [
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-line",
                    "valueInteger": 1
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-col",
                    "valueInteger": 6767
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-issue-source",
                    "valueString": "InstanceValidator"
                },
                {
                    "url": "http://hl7.org/fhir/StructureDefinition/operationoutcome-message-id",
                    "valueCode": "BUNDLE_BUNDLE_ENTRY_REVERSE_R4"
                }
            ],
            "severity": "warning",
            "code": "invalid",
            "diagnostics": "Entry 'http://test.fhir.ch/r4/PractitionerRole/FamilienHausarztAtHausarzt' isn't reachable by traversing forwards from the Composition. Only Provenance is approved to be used this way (R4 section 3.3.1)",
            "expression": [
                "Bundle.entry[9]"
            ]
        }
    ]
}
```

**Comment by oliveregger** [comment]
Bundle.entry[0]: Bundle.entry[0].resource.section[0].entry[0]: Bundle.entry[7].resource/*AllergyIntolerance/AllergyToCatDander*/.code: Error from http://tx.fhir.org/r4: Error: A definition for the value Set 'http://fhir.ch/ig/ch-allergyintolerance/ValueSet/CHAllergyIntoleranceValueSet' could not be found

**Comment by oliveregger** [comment]
could it be a problem for the tx server that http://fhir.ch/ig/ch-allergyintolerance/ValueSet/CHAllergyIntoleranceValueSet is now located in ch-term package?

**Comment by oliveregger** [comment]
fixed with update to core version 6.3.11

**Comment by oliveregger** [comment]
fixed in v3.8.6
